packages/components/date-picker/src/date-picker-com/panel-date-pick.vue

```diff
@@ -405,8 +406,37 @@ const keyboardMode = computed<string>(() => {
 
 const hasShortcuts = computed(() => !!shortcuts.length)
 
+const getValidDateOfMonth = (month: number) => {
+  const _value = innerDate.value.startOf('month').month(month)
+  const dateCountOfMonth = _value.daysInMonth()
+  for (let i = 0; i < dateCountOfMonth; i++) {
+    const _countDate = _value.add(i, 'day')
+    if (!disabledDate?.(_countDate.toDate())) {
+      return _countDate
+    }
+  }
+  return _value
+}
+
+const getValidDateOfYear = (year: number) => {
+  innerDate.value = innerDate.value.year(year)
+  if (!disabledDate?.(innerDate.value.toDate())) {
+    return innerDate.value
+  }
+  const month = innerDate.value.month()
+  if (!datesInMonth(year, month, lang.value).every(disabledDate)) {
+    return getValidDateOfMonth(month)
+  }
+  for (let i = 0; i < 12; i++) {
+    if (!datesInMonth(year, i, lang.value).every(disabledDate)) {
+      return getValidDateOfMonth(i)
+    }
+  }
+  return innerDate.value
+}
+
```

Comment:
Perhaps these two methods can also be extracted into `utils`.

Reply:
In this case, disabledDate, innerDate, and lang need to be parameters in the current function. I don't think it's very good.

Reply:
Ok.